src/store/ShareStore.js

```diff
@@ -67,19 +64,23 @@ export default class ShareStore {
         hashPromise
             .then((hash) => {
```

Comment:
How about...

```js
hasPromise
    .then(this.onHashAvailable)
    .catch(this.onHashError);
```

I'm not sure whether you can use [`runInAction`](https://mobx.js.org/refguide/action.html#-runinaction-name-thunk) but I think this is pretty readable to me.

Reply:
@brucebinary amazing solution